### ISSUE TITLE
Add Issue Management to Visual Studio

### ISSUE THREAD

**Issue by drobertson123** [github/VisualStudio#1482]
Love this extension, but it has a big missing item. It would be extremely helpful to be able to manage GitHub issues from VS while I am working on the code.

Yes, I know I can jump out to the web page and do this, but it is a major slowdown and kills workflow. Having Issues Management inside Visual Studio would be much better.

The key features would be;

- Create new issue from inside VS
- Issue management, including tags and open/close
- Add comments to Issue
- Easy Issue attribution for commits (a lookup or list, no more figuring our what issue number by going to github)

Wishlist Items;

- [In code editor] Highlight code and Add as Issue Comment or Create Issue
- [not sure how this would be implemented] Code links in issues, go from issue directly to code area

Keep up the good work and thank you.

**Comment by grokys** [comment]
Hi @drobertson123 - thanks for the feature request. You will be pleased to know that we plan to start on support for Issues after we've finished with Pull Request Reviews, which will hopefully be within the next few months. I'll come back here to check on your key features before starting that work.

**Comment by drobertson123** [comment]
@grokys I just wanted to check in and see if the "Issues" features timeline was getting any closer.

I would be happy to add any details or ideas to the thoughts above if that is useful. Otherwise, I am happy to get out of the way and let you work.   :-)

**Comment by drobertson123** [comment]
@grokys I just wanted to check in on the "Issues Management" features for the Visual Studio add-in.  Any idea if this is on the projects list anytime soon.

I would be happy to write up more detailed notes on this if you would like.

**Comment by grokys** [comment]
Hey again @drobertson123! Yes this is on our roadmap for the next quarter - we're hoping to have an MVP in that timeline. Any ideas/detailed notes would be welcomed!

**Comment by drobertson123** [comment]
@grokys any movement on the Issues management issue?

**Comment by drobertson123** [comment]
Thank you for the update @meaghanlewis . 

It is just starting to get pretty frustrating waiting on this. Issues management is a day in and day out task. I deal with issues far more often than pull requests. It may not be sexy, but it is critical and one of the most frustrating parts of using GitHub for Visual Studio.

I am just asking that the mundane, day to day tools don't get ignored to chase after cooler toys that don't really help workflow as much. Please don't forget about Issues management. I keep hearing next quarter, next quarter. Please actually get it on the production schedule.

**Comment by grokys** [comment]
Hi Doug,

Apologies for the delay on this, we're a very small team but rest assured we're not sitting here twiddling our thumbs.

We did a spike of Issue functionality and determined that the infrastructure we had in place for PRs was too slow the large number issues that many projects have (most projects have many more issues than PRs). This necessitated a move to GraphQL (including writing a .NET client library for GraphQL: https://github.com/grokys/Octokit.GraphQL) and a rewrite of  our PR list to use data virtualization (#1737 -- a feature that isn't natively supported in WPF).

#1737 is just about finished now, so given that the PR list and the issue list will share 90% of their code, adding issues shouldn't be too much work now (famous last words). However, we do have other areas of improvement that we receive far more feature and support requests for than issues, so I can't guarantee we'll get to it _immediately_.

**Comment by drobertson123** [comment]
I do appreciate the work you do and understand the limitations. Sorry if I came off as demanding, but this has been a constant source of frustration and directly affects day to day workflow. 

Good luck finishing [#1737](https://github.com/github/VisualStudio/pull/1737) and I look forward to seeing the results.

I just looked at [Octokit.GraphQL](https://github.com/grokys/Octokit.GraphQL). It looks very useful. Thanks.

**Comment by efcarey** [comment]
Any movement on this update? The bare minimum would be to simply view issues in a list/grid so you can use the issues numbers when making PRs. This would make my devs never have to leave visual studio and a great feature. Thanks for your work!

**Comment by meaghanlewis** [comment]
👋 @efcarey no, there is currently no update on when we will finish implementing this feature. There is still a WIP pull request for issue implementation here: https://github.com/github/VisualStudio/pull/1957.

Our goal is to make devs never have to leave Visual Studio either, but as mentioned above, we are just a small core team of maintainers working the best we can to move toward that goal. 😄

**Comment by Kalter111** [comment]
Hi! Is there any news about this feature?

**Comment by Shujee** [comment]
Just wanted to see if this is still on your roadmap, now that 4 years have passed?

**Comment by danielkornev** [comment]
same question! There used to be a good plugin for VS in the past but it died after 2015: https://github.com/rprouse/GitHubExtension
